src/token/common/erc2981/erc2981.cairo

```diff
@@ -0,0 +1,165 @@
+        /// Defaults to 10000 so fees are expressed in basis points
+        fn _fee_denominator(self: @ComponentState<TContractState>) -> u256 {
+            10000
+        }
+
+        /// Returns the royalty information that all ids in this contract will default to.
+        fn _default_royalty(
+            self: @ComponentState<TContractState>
+        ) -> (ContractAddress, u256, u256) {
+            let royalty_info: RoyaltyInfo = self.default_royalty_info.read();
+            (royalty_info.receiver, royalty_info.royalty_fraction, self._fee_denominator())
+        }
+
+        /// Sets the royalty information that all ids in this contract will default to.
+        ///
+        /// Requirements:
+        ///
+        /// - `receiver` cannot be the zero address.
+        /// - `fee_numerator` cannot be greater than the fee denominator.
+        fn _set_default_royalty(
+            ref self: ComponentState<TContractState>,
+            receiver: ContractAddress,
+            fee_numerator: u256,
+        ) {
+            let denominator = self._fee_denominator();
+            assert!(fee_numerator <= denominator, "Invalid default royalty");
+            assert!(!receiver.is_zero(), "Invalid default royalty receiver");
+            self
+                .default_royalty_info
+                .write(RoyaltyInfo { receiver, royalty_fraction: fee_numerator })
+        }
+
+
+        /// Sets the royalty information for a specific token id, overriding the global default.
+        ///
+        /// Requirements:
+        ///
+        /// - `receiver` cannot be the zero address.
+        /// - `fee_numerator` cannot be greater than the fee denominator.
+        fn _set_token_royalty(
+            ref self: ComponentState<TContractState>,
+            token_id: u256,
+            receiver: ContractAddress,
+            fee_numerator: u256
+        ) {
+            let denominator = self._fee_denominator();
+            assert!(fee_numerator <= denominator, "Invalid token royalty");
+            assert!(!receiver.is_zero(), "Invalid token royalty receiver");
+
+            self
+                .token_royalty_info
+                .write(token_id, RoyaltyInfo { receiver, royalty_fraction: fee_numerator },)
+        }
+
+        /// Returns the royalty information that all ids in this contract will default to.
+        fn _token_royalty(
+            self: @ComponentState<TContractState>, token_id: u256
+        ) -> (ContractAddress, u256, u256) {
+            let royalty_info: RoyaltyInfo = self.token_royalty_info.read(token_id);
+            let mut receiver = royalty_info.receiver;
+            let mut royalty_fraction = royalty_info.royalty_fraction;
+
+            if receiver.is_zero() {
+                let default_royalty_info: RoyaltyInfo = self.default_royalty_info.read();
+                receiver = default_royalty_info.receiver;
+                royalty_fraction = default_royalty_info.royalty_fraction;
+            };
```

Comment:
Same thing here, no need to have mutable variables, if-else loop should be preferred

Reply:
Code updated